### ISSUE TITLE
Update ST-Util to newer version

### ISSUE THREAD

**Issue by c-grant** [AdaCore/gps#119]
Not sure if this is the right place for this issue

It looks like the version of st-util  included in GNAT Studio is 1.3.1. This does not support the new ST-link V3 dev boards. The STLINK-org has been upgraded to support this newer interface.

**Comment by JCGobbi** [comment]
The GNAT Studio doesn't install ST-Link utilities, it uses the version installed on your system (that must be in your PATH). So you need to install a newer version on your system to update the installed ST-Link.

**Comment by c-grant** [comment]
It appears to be installed with the Adacore Arm tools for Gnatstudio to use, so not with Gnatstudio proper. Is there a better place to submit this ticket?

**Comment by JCGobbi** [comment]
This is the suitable place to put this ticket because flashing files inside GNAT Studio uses ST-Link tools, but Adacore doesn't furnish them, neither with the ARM compiler. If you didn't have it installed on your system, the GNAT Studio would report that it didn't find st-link.
If you are using Linux, open a terminal and try the command "find / -name st-flash" or use the file manager "search" to get a list of directories where the ST-Link tools are installed. If you are using Windows, search st-link*.exe with the file manager.
If you are a beginner with Ada, take a look at https://github.com/JCGobbi/Documentation/tree/master/Ada, where you will find good manuals for starting.

**Comment by c-grant** [comment]
I am running it on Windows and have taught a number of classes with it. 

Using the executable gnat-2021-20210519-arm-elf-windows64-bin.exe provided from the community download links it indeed does get installed.

The files are located in the following folder with a fresh install C:\GNAT\2021-arm-elf\bin of the Arm toolchain. 

see attached image

![image](https://user-images.githubusercontent.com/60671494/130681059-97baa365-5b0f-4efa-b738-a1c38e4c4bb7.png)

**Comment by JCGobbi** [comment]
Yes, you are correct, the 2021 ARM toolchain installs the ST-Link tools.
Perhaps you may get the newer files compiled for Windows from https://github.com/stlink-org/stlink/releases/tag/v1.7.0 (choosing between i686 or x86_64) and unzip it in any directory and try the following:
1. Copy the files st-flash.exe, st-info.exe and st-util.exe that are in the C:\GNAT\2021-arm-elf\bin to a backup directory.
2. Copy the files libstlink.a, libstlink-shared.dll and libstlink-shared.dll.a that are in the C:\GNAT\2021-arm-elf\lib to a backup directory.
3. Copy the files st-flash.exe, st-info.exe and st-util.exe from the new installation into C:\GNAT\2021-arm-elf\bin, with overwrite.
4. Copy the files libstlink.a, libstlink.dll and libstlink.dll.a from the new installation into C:\GNAT\2021-arm-elf\lib, and delete the old files.
5. If all goes wrong, get the backup files and overwrite the new ones. But I think it will work.
Good luck!

**Comment by AnthonyLeonardoGracio** [comment]
Hello,

We furnished our own version of ST-Link utilities because of bugs with their version. We'll see if this has been solved since then.

Regards,
